FreeRTOS/FreeRTOS-Kernel
# Remove  that Heap_1.c unnecessary judgment and code logic

Remove  that Heap_1.c unnecessary judgment and code logic
-----------


- The original code is as follows

```c
 /* Byte alignment required. Check for overflow. */
if( ( xWantedSize + ( portBYTE_ALIGNMENT - ( xWantedSize & portBYTE_ALIGNMENT_MASK ) ) ) > xWantedSize )
{
    xWantedSize += ( portBYTE_ALIGNMENT - ( xWantedSize & portBYTE_ALIGNMENT_MASK ) );
}
else
{
    xWantedSize = 0;
}
```

- This is equivalent to the following code
```c
//Step 0: The original code is as follows
if( ( xWantedSize + ( portBYTE_ALIGNMENT - ( xWantedSize & portBYTE_ALIGNMENT_MASK ) ) ) > xWantedSize )
//Step 1: Remove the variable xWantedSize
if( (( portBYTE_ALIGNMENT - ( xWantedSize & portBYTE_ALIGNMENT_MASK ) ) ) > 0 )
//Step 2:Move the subtraction part after the >
if( ( portBYTE_ALIGNMENT ) > ( xWantedSize & portBYTE_ALIGNMENT_MASK ) )
/*
* Step 3:Now, let's evaluate the conditional statement
* xWantedSize & portBYTE_ALIGNMENT_MASK  executed by this statement must range from [0,portBYTE_ALIGNMENT_MASK]
* We can find all possible portBYTE_ALIGNMENT_MASK values in portle.h
* It follows that portBYTE_ALIGNMENT must be greater than portBYTE_ALIGNMENT_MASK
*/
#if portBYTE_ALIGNMENT == 32
    #define portBYTE_ALIGNMENT_MASK    ( 0x001f )
#elif portBYTE_ALIGNMENT == 16
    #define portBYTE_ALIGNMENT_MASK    ( 0x000f )
#elif portBYTE_ALIGNMENT == 8
    #define portBYTE_ALIGNMENT_MASK    ( 0x0007 )
#elif portBYTE_ALIGNMENT == 4
    #define portBYTE_ALIGNMENT_MASK    ( 0x0003 )
#elif portBYTE_ALIGNMENT == 2
    #define portBYTE_ALIGNMENT_MASK    ( 0x0001 )
#elif portBYTE_ALIGNMENT == 1
    #define portBYTE_ALIGNMENT_MASK    ( 0x0000 )
#else /* if portBYTE_ALIGNMENT == 32 */
    #error "Invalid portBYTE_ALIGNMENT definition"
#endif /* if portBYTE_ALIGNMENT == 32 */
```
- So the statement can be omitted, and there is no such thing as an overflow caused by byte alignment.

Test Steps
-----------

None.

Checklist:
----------



Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
